agent/taskresource/interface.go

```diff
@@ -35,6 +37,10 @@ type TaskResource interface {
 	SetCreatedAt(time.Time)
 	// GetCreatedAt sets the timestamp for resource's creation time
 	GetCreatedAt() time.Time
+	// Create performs resource creation
+	Create(task *api.Task)
```

Comment:
No return value here?

Reply:
Thanks, missed it. shall add here and in `Cleanup`.